src/Cms/ModelWithContent.php

```diff
@@ -842,9 +860,24 @@ public function uuid(): Uuid|null
 	 */
 	public function writeContent(array $data, string $languageCode = null): bool
 	{
-		return Data::write(
-			$this->contentFile($languageCode),
-			$this->contentFileData($data, $languageCode)
-		);
+		$data     = $this->contentFileData($data, $languageCode);
+		$language = $this->storage()->languageCodeToObject($languageCode);
+
+		$identifier = $this::CLASS_ALIAS === 'page' && $this->isDraft() === true ?
+			VersionIdentifier::changes() :
+			VersionIdentifier::published();
+
+		// we can only update if the version already exists
+		if ($this->storage()->exists($identifier, $language) === true) {
+			$this->storage()->update($identifier, $language, $data);
+			return true;
+		}
```

Comment:
It already does, but then it throws an exception (see my comments above for the reasoning). Having an uncaught exception here would break this method, so we need the `exists()` check.

Actually it would be more robust to replace the `exists()` check with a `try catch` block for `NotFoundException`, but I'm not sure if that's too much.

Reply:
Try-catch would match `readContent`, so maybe that'd be more consistent then